imcsdk/apis/admin/ldap.py

```diff
@@ -0,0 +1,549 @@
+        ldap_role_group_get(handle, domain='abcd.pqrs.com', name='abcd')
+    """
+
+    mos = handle.query_classid('AaaLdapRoleGroup')
+    for mo in mos:
+        if mo.domain == domain and mo.name == name:
+            return mo
+
+    return None
+
+
+def ldap_role_group_exists(handle, **kwargs):
+    """
+    Checks if the ldap group exists with the specified settings
+
+    Args:
+        handle (ImcHandle)
+        kwargs: Key-Value paired arguments for future use
+
+    Returns:
+        (True, AaaLdapRoleGroup) if match found, else (False, None)
+
+    Examples:
+        ldap_role_group_exists(handle, domain='abcd.pqrs.com', name='abcd', role='user')
+    """
+    domain = kwargs.get('domain')
+    name = kwargs.get('name')
+    mo = ldap_role_group_get(handle, domain=domain, name=name)
+    if not mo:
+        return False, None
+
+    if _is_valid_arg('role', kwargs):
+        if mo.role != kwargs.get('role'):
+            return False, None
+
+    return True, mo
+
+
+def ldap_role_group_delete(handle, domain, name, **kwargs):
+    """
+    Deletes the ldap role group based on domain and name
+
+    Args:
+        handle (ImcHandle)
+        domain (str): The LDAP server domain the group resides in.
+        name (str): The name of the group in the LDAP server database.
+        kwargs: Key-Value paired arguments for future use
+
+    Returns:
+        None
+
+    Examples:
+        ldap_role_group_delete(handle, domain='abcd.pqrs.com', name='abcd')
+    """
+
+    mo = ldap_role_group_get(handle, domain=domain, name=name, **kwargs)
+    if mo is None:
+        raise ImcOperationError("Delete LDAP Role Group",
+                                "Role Group Not found")
+
+    mo.admin_action = 'clear'
+    handle.set_mo(mo)
+
+
+def ldap_certificate_management_enable(handle):
+    """
+    Enables ldap certificate management
+
+    Args:
+        handle (ImcHandle)
+
+    Returns:
+        LdapCACertificateManagement object
+    """
+    mo = _get_mo(handle, dn="sys/ldap-ext/ldap-ca-cert-mgmt")
+    mo.binding_certificate = "enabled"
+    handle.set_mo(mo)
+    return handle.query_dn(mo.dn)
+
+
+def ldap_certificate_management_disable(handle):
+    """
+    Disables ldap certificate management
+
+    Args:
+        handle (ImcHandle)
+
+    Returns:
+        LdapCACertificateManagement object
+    """
+    mo = _get_mo(handle, dn="sys/ldap-ext/ldap-ca-cert-mgmt")
+    mo.binding_certificate = "disabled"
+    handle.set_mo(mo)
+    return handle.query_dn(mo.dn)
+
+
+def is_ldap_certificate_management_enabled(handle):
+    """
+    Checks if LDAP certificate management is enabled
+
+    Args:
+        handle (ImcHandle)
+
+    Returns:
+        bool
+    """
+    mo = _get_mo(handle, dn="sys/ldap-ext/ldap-ca-cert-mgmt")
+    return mo.binding_certificate.lower() == "enabled"
+
+
+def ldap_certificate_download(handle, user=None, pwd=None, remote_server=None,
+                              remote_file=None, protocol='tftp', **kwargs):
```

Comment:
remote_server, remote_file is always required. right? If so, it should not have `None` as default. user/pwd may not be required for `http` case i guess.